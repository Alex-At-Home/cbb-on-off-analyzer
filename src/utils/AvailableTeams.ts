
export type AvailableTeamMeta = {
  team: string,
  year: string,
  gender: string,
  index_template: string
}

export class AvailableTeams {

  private static flatten(
    arr: Array<Array<AvailableTeamMeta>>
  ): Array<AvailableTeamMeta> {
    return arr.reduce(function(acc, v) {
      return acc.concat(v);
    }, [] as Array<AvailableTeamMeta>);
  }

  static getTeams(
    team: string | null, year: string | null, gender: string | null
  ): Array<AvailableTeamMeta> {
    const list = team ?
      (AvailableTeams.byName[team] || ([] as Array<AvailableTeamMeta>)) :
      AvailableTeams.flatten(Object.values(AvailableTeams.byName));

    return list.filter(function(record) {
      return (!team || (team == record.team)) &&
        (!year || (year == record.year)) && (!gender || (gender == record.gender));
    });
  }
  static getTeam(
    team: string, year: string, gender: string
  ): AvailableTeamMeta | null {
      const retVal = AvailableTeams.getTeams(team, year, gender);
      return (retVal.length > 0) ? retVal[0] : null;
  }
  static readonly byName: Record<string, Array<AvailableTeamMeta>> = {
    // Maryland!
    "Maryland": [
      { team: "Maryland", year: "2016/7", gender: "Men", index_template: "maryland" },
      { team: "Maryland", year: "2018/9", gender: "Men", index_template: "misc_conf" },
      { team: "Maryland", year: "2018/9", gender: "Women", index_template: "women_bigten" },
    ],
    // Other BIG
    "Indiana": [
      { team: "Indiana", year: "2018/9", gender: "Men", index_template: "bigten" },
    ],
    "Nebraska": [
      { team: "Nebraska", year: "2018/9", gender: "Men", index_template: "bigten" },
    ],
    "Wisconsin": [
      { team: "Wisconsin", year: "2018/9", gender: "Men", index_template: "bigten" },
    ],
    "Ohio St.": [
      { team: "Ohio St.", year: "2018/9", gender: "Men", index_template: "bigten" },
    ],
    "Michigan St.": [
      { team: "Michigan St.", year: "2018/9", gender: "Men", index_template: "bigten" },
    ],
    "Iowa": [
      { team: "Iowa", year: "2018/9", gender: "Men", index_template: "bigten" },
    ],
    "Michigan": [
      { team: "Michigan", year: "2018/9", gender: "Men", index_template: "bigten" },
    ],
    "Purdue": [
      { team: "Purdue", year: "2018/9", gender: "Men", index_template: "bigten" },
    ],
    "Illinois": [
      { team: "Illinois", year: "2018/9", gender: "Men", index_template: "bigten" },
    ],
    "Penn St.": [
      { team: "Penn St.", year: "2018/9", gender: "Men", index_template: "bigten" },
    ],
    "Northwestern": [
      { team: "Northwestern", year: "2018/9", gender: "Men", index_template: "bigten" },
    ],
    "Rutgers": [
      { team: "Rutgers", year: "2018/9", gender: "Men", index_template: "bigten" },
    ],
    "Minnesota": [
      { team: "Minnesota", year: "2018/9", gender: "Men", index_template: "bigten" },
    ],
    // Misc
    "Cincinnati": [
      { team: "Cincinnati", year: "2018/9", gender: "Men", index_template: "misc_conf" },
    ],
    "Kentucky": [
      { team: "Kentucky", year: "2018/9", gender: "Men", index_template: "misc_conf" },
    ],
    "Louisville": [
      { team: "Louisville", year: "2018/9", gender: "Men", index_template: "misc_conf" },
    ],
    "Saint Mary's (CA)": [
      { team: "Saint Mary's (CA)", year: "2018/9", gender: "Men", index_template: "misc_conf" },
    ],
    "Virginia": [
      { team: "Virginia", year: "2018/9", gender: "Men", index_template: "misc_conf" },
    ],
    "UConn": [
      { team: "UConn", year: "2018/9", gender: "Men", index_template: "misc_conf" },
    ],
  }
}
