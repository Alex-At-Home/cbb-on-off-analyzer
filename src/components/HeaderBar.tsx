// React imports:
import React, { useState, useEffect } from 'react';
import Router, { useRouter } from 'next/router';
import Link from 'next/link';

// Next imports:
import { NextPage } from 'next';

// Lodash:
import _ from "lodash";

// Bootstrap imports:
import 'bootstrap/dist/css/bootstrap.min.css';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
import Popover from 'react-bootstrap/Popover';
import Tooltip from 'react-bootstrap/Tooltip';

// App components:
import { getCommonFilterParams, ParamPrefixes, CommonFilterParams, GameFilterParams, LineupFilterParams, TeamReportFilterParams } from '../utils/FilterModels';

// Utils:
import { UrlRouting } from "../utils/UrlRouting";
import { HistoryManager } from '../utils/HistoryManager';

type Props = {
  thisPage: string,
  common: CommonFilterParams
};

const HeaderBar: React.FunctionComponent<Props> = ({thisPage, common}) => {

  const server = (typeof window === `undefined`) ? //(ensures SSR code still compiles)
    "server" : window.location.hostname

  // Last visited
  function getLastGameUrl() {
    return UrlRouting.getGameUrl(
      UrlRouting.removedSavedKeys(
        HistoryManager.getLastQuery(ParamPrefixes.game) || ""
      ) as GameFilterParams, {});
  }
  function getLastLineupUrl() {
    return UrlRouting.getLineupUrl(
      UrlRouting.removedSavedKeys(
        HistoryManager.getLastQuery(ParamPrefixes.lineup) || ""
      ) as LineupFilterParams, {});
  }
  function getLastReportUrl() {
    return UrlRouting.getTeamReportUrl(
      UrlRouting.removedSavedKeys(
        HistoryManager.getLastQuery(ParamPrefixes.report) || ""
      ) as TeamReportFilterParams);
  }
  // From baseline query
  function getBaseGameUrl() {
    return UrlRouting.getGameUrl(
      getCommonFilterParams(common)  as GameFilterParams, {}
    );
  }
  function getBaseLineupUrl() {
    return UrlRouting.getLineupUrl(
      getCommonFilterParams(common) as LineupFilterParams, {}
    );
  }
  function getBaseReportUrl() {
    return UrlRouting.getTeamReportUrl(
      getCommonFilterParams(common) as TeamReportFilterParams
    );
  }

  function maybeShowBlog() {
    if (!_.startsWith(server, "cbb-on-off-analyzer")) {
      const blogTooltip = (
        <Tooltip id="blogTooltip">Articles describing how to use the tool</Tooltip>
      );
      return <Col xs={1} className="small">
        <OverlayTrigger placement="auto" overlay={blogTooltip}>
          <a href="https://hoop-explorer.blogspot.com/p/blog-page.html" target="_new">Blog...</a>
        </OverlayTrigger>
      </Col>;
    }
  }

  const baseGameTooltip = (
    <Tooltip id="baseGameTooltip">Go to the On/Off Analysis page with the current baseline query</Tooltip>
  );
  const lastGameTooltip = (
    <Tooltip id="lastGameTooltip">Go back to the most recently submitted On/Off Analysis page</Tooltip>
  );
  const baseLineupTooltip = (
    <Tooltip id="baseLineupTooltip">Go to the Lineup Analysis page with the current baseline query</Tooltip>
  );
  const lastLineupTooltip = (
    <Tooltip id="lastLineupTooltip">Go back to the most recently submitted Lineup Analysis page</Tooltip>
  );
  const baseReportTooltip = (
    <Tooltip id="baseReportTooltip">Go to the On/Off Report page with the current baseline query</Tooltip>
  );
  const lastReportTooltip = (
    <Tooltip id="lastReportTooltip">Go back to the most recently submitted On/Off Report page</Tooltip>
  );
  const chartTooltip = (
    <Tooltip id="lastReportTooltip">View a gallery of interesting basketball analytics charts</Tooltip>
  );

  //(only render client-side - was running into cache issues of the Link href)
  return (typeof window !== `undefined`) ? <Container>
      <Row className="border-top">
        {(thisPage != ParamPrefixes.game) ?
            <Col className="text-center small">
              <OverlayTrigger placement="auto" overlay={baseGameTooltip}>
                <span><Link href={getBaseGameUrl()}><a>On/Off: Base</a></Link></span>
              </OverlayTrigger>
            </Col> : null
        }
        {(thisPage != ParamPrefixes.game) ?
            <Col className="text-center small">
              <OverlayTrigger placement="auto" overlay={lastGameTooltip}>
                <span><Link href={getLastGameUrl()}><a>On/Off: Last</a></Link></span>
              </OverlayTrigger>
            </Col> : null
        }
        {(thisPage != ParamPrefixes.lineup) ?
            <Col className="text-center small">
              <OverlayTrigger placement="auto" overlay={baseLineupTooltip}>
                <span><Link href={getBaseLineupUrl()}><a>Lineups: Base</a></Link></span>
              </OverlayTrigger>
            </Col> : null
        }
        {(thisPage != ParamPrefixes.lineup) ?
            <Col className="text-center small">
            <OverlayTrigger placement="auto" overlay={lastLineupTooltip}>
                <span><Link href={getLastLineupUrl()}><a>Lineups: Last</a></Link></span>
              </OverlayTrigger>
            </Col> : null
        }
        {(thisPage != ParamPrefixes.report) ?
            <Col className="text-center small">
            <OverlayTrigger placement="auto" overlay={baseReportTooltip}>
                <span><Link href={getBaseReportUrl()}><a>Report: Base</a></Link></span>
            </OverlayTrigger>
            </Col> : null
        }
        {(thisPage != ParamPrefixes.report) ?
            <Col className="text-center small">
              <OverlayTrigger placement="auto" overlay={lastReportTooltip}>
                <span><Link href={getLastReportUrl()}><a>Report: Last</a></Link></span>
              </OverlayTrigger>
            </Col> : null
        }
        {(thisPage != "charts") ?
            <Col className="text-center small">
              <OverlayTrigger placement="auto" overlay={chartTooltip}>
                <span><Link href={"/Charts"}><a>Charts</a></Link></span>
              </OverlayTrigger>
            </Col> : null
        }
        {maybeShowBlog()}
      </Row>
    </Container> : null;
}

export default HeaderBar;
