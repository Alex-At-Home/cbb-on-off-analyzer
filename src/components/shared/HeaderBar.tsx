// React imports:
import React, { useState, useEffect } from 'react';
import Router, { useRouter } from 'next/router';
import Link from 'next/link';

// Next imports:
import { NextPage } from 'next';

// Lodash:
import _ from "lodash";

// Bootstrap imports:
import 'bootstrap/dist/css/bootstrap.min.css';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import OverlayTrigger from 'react-bootstrap/OverlayTrigger';
import Popover from 'react-bootstrap/Popover';
import Tooltip from 'react-bootstrap/Tooltip';
import Dropdown from 'react-bootstrap/Dropdown';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAngleDown } from '@fortawesome/free-solid-svg-icons';


// Utils:
import { getCommonFilterParams, ParamPrefixes, CommonFilterParams, GameFilterParams, LineupFilterParams, TeamReportFilterParams } from '../../utils/FilterModels';
import { UrlRouting } from "../../utils/UrlRouting";
import { HistoryManager } from '../../utils/HistoryManager';

type Props = {
  thisPage: string,
  common: CommonFilterParams,
  override?: boolean //(for testing)
};

/** Dropdowns for controlling navigation */
const StyledDropdown = React.forwardRef<HTMLAnchorElement>((props, ref) => (
    <a
      href=""
      ref={ref}
      onClick={(e) => {
        e.preventDefault();
        (props as any).onClick(e);
      }}
    >
        {props.children}
        &nbsp;&nbsp;<FontAwesomeIcon icon={faAngleDown} />
    </a>
));

const HeaderBar: React.FunctionComponent<Props> = ({thisPage, common, override}) => {

  const server = (typeof window === `undefined`) ? //(ensures SSR code still compiles)
    "server" : window.location.hostname;

  // Last visited
  function getLastGameUrl() {
    return UrlRouting.getGameUrl(
      UrlRouting.removedSavedKeys(
        HistoryManager.getLastQuery(ParamPrefixes.game) || ""
      ) as GameFilterParams, {});
  }
  function getLastLineupUrl() {
    return UrlRouting.getLineupUrl(
      UrlRouting.removedSavedKeys(
        HistoryManager.getLastQuery(ParamPrefixes.lineup) || ""
      ) as LineupFilterParams, {});
  }
  function getLastReportUrl() {
    return UrlRouting.getTeamReportUrl(
      UrlRouting.removedSavedKeys(
        HistoryManager.getLastQuery(ParamPrefixes.report) || ""
      ) as TeamReportFilterParams);
  }
  // From baseline query
  function getBaseGameUrl() {
    return UrlRouting.getGameUrl(
      getCommonFilterParams(common) as GameFilterParams, {}
    );
  }
  function getBaseLineupUrl() {
    return UrlRouting.getLineupUrl(
      getCommonFilterParams(common) as LineupFilterParams, {}
    );
  }
  function getBaseReportUrl() {
    return UrlRouting.getTeamReportUrl(
      getCommonFilterParams(common) as TeamReportFilterParams
    );
  }

  const baseGameTooltip = (
    <Tooltip id="baseGameTooltip">Go to the On/Off Analysis page with the current baseline query</Tooltip>
  );
  const lastGameTooltip = (
    <Tooltip id="lastGameTooltip">Go back to the most recently submitted On/Off Analysis page</Tooltip>
  );
  const baseLineupTooltip = (
    <Tooltip id="baseLineupTooltip">Go to the Lineup Analysis page with the current baseline query</Tooltip>
  );
  const lastLineupTooltip = (
    <Tooltip id="lastLineupTooltip">Go back to the most recently submitted Lineup Analysis page</Tooltip>
  );
  const baseReportTooltip = (
    <Tooltip id="baseReportTooltip">Go to the On/Off Report page with the current baseline query</Tooltip>
  );
  const lastReportTooltip = (
    <Tooltip id="lastReportTooltip">Go back to the most recently submitted On/Off Report page</Tooltip>
  );
  const chartTooltip = (
    <Tooltip id="lastReportTooltip">View a gallery of interesting basketball analytics charts</Tooltip>
  );

  /** Adds the current selection to history before navigating away */
  const onNav = (e: any) => {
    if (!_.isEmpty(common)) { //(do nothing if the page has just been loaded and not modified)
      const key = UrlRouting.getUrl({ [UrlRouting.noSuffix]: common });
      HistoryManager.addParamsToHistory(key, thisPage);
    }
  };

  /** Builds a nice looking nav dropdown item */
  const buildNavItem = (itemName: string, tooltip: React.ReactElement<any>, url: string) => {
    return <OverlayTrigger rootClose placement="left" overlay={tooltip}>
      <span>
        <Link href={url}>
          <div>
            <a className="text-center small" href={url} onClick={onNav}>{itemName}</a>
          </div>
        </Link>
      </span>
    </OverlayTrigger>
  };

  const dropdownStyle = {
    width:"40px", left:"50%", marginLeft:"-20px"
  };

  /** Build a nice looking nav dropdown */
  const buildNavDropdown = (
      name: string,
      baseTooltip: React.ReactElement<any>, baseUrl: string,
      lastTooltip: React.ReactElement<any>, lastUrl: string
  ) => {
    //(mega grovelling with types required to get TS to compile with example from react bootstrap custom dropdown example code)
    return <Dropdown
    >
        <Dropdown.Toggle id={baseUrl} as={StyledDropdown as unknown as undefined}>{name}</Dropdown.Toggle>
        <Dropdown.Menu style={dropdownStyle}>
          <Dropdown.Item>
            {buildNavItem("Base", baseTooltip, baseUrl)}
          </Dropdown.Item>
          <Dropdown.Item>
            {buildNavItem("Last", lastTooltip, lastUrl)}
          </Dropdown.Item>
        </Dropdown.Menu>
      </Dropdown>;
  };

  /** Build a nice looking nav dropdown */
  const buildChartDropdown = () => {
    //(mega grovelling with types required to get TS to compile with example from react bootstrap custom dropdown example code)
    return <Dropdown>
        <Dropdown.Toggle id="chartDropDown" as={StyledDropdown as unknown as undefined}>Charts</Dropdown.Toggle>
        <Dropdown.Menu style={dropdownStyle}>
          <Dropdown.Item>
            {buildNavItem("Player Positions", chartTooltip, "/Charts")}
          </Dropdown.Item>
        </Dropdown.Menu>
      </Dropdown>;
  };

  /** Show blog if rendering external version of the page */
  function maybeShowBlog() {
    if (!_.startsWith(server, "cbb-on-off-analyzer")) {
      const blogTooltip = (
        <Tooltip id="blogTooltip">Articles describing how to use the tool</Tooltip>
      );
      return <Col xs={1} className="small">
        <OverlayTrigger placement="auto" overlay={blogTooltip}>
          <a href="https://hoop-explorer.blogspot.com/p/blog-page.html" target="_new">Blog...</a>
        </OverlayTrigger>
      </Col>;
    }
  }

  //(only render client-side - was running into cache issues of the Link href)
  return (override || (typeof window !== `undefined`)) ? <Container>
      <Row className="border-top">
        {(thisPage != ParamPrefixes.game) ?
            <Col className="text-center small">
              {buildNavDropdown("On/Off", baseGameTooltip, getBaseGameUrl(), lastGameTooltip, getLastGameUrl())}
            </Col> : null
        }
        {(thisPage != ParamPrefixes.lineup) ?
            <Col className="text-center small">
              {buildNavDropdown("Lineups", baseLineupTooltip, getBaseLineupUrl(), lastLineupTooltip, getLastLineupUrl())}
            </Col> : null
        }
        {(thisPage != ParamPrefixes.report) ?
          <Col className="text-center small">
            {buildNavDropdown("Reports", baseReportTooltip, getBaseReportUrl(), lastReportTooltip, getLastReportUrl())}
          </Col> : null
        }
        {(thisPage != "charts") ?
          <Col className="text-center small">
            {buildChartDropdown()}
          </Col> : null
        }
        {maybeShowBlog()}
      </Row>
    </Container> : null;
}

export default HeaderBar;
