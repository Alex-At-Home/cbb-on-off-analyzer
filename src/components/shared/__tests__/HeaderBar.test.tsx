/**
 * @jest-environment jsdom
 */

import React from 'react';
import HeaderBar from '../HeaderBar';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import { ParamPrefixes} from "../../../utils/FilterModels";


describe("HeaderBar", () => {
  test("HeaderBar - should create snapshot (game)", () => {
    const component = renderer.create(<HeaderBar
      thisPage={ParamPrefixes.game}
      common={{team:"Test1", gender:"Men", year:"2019/20", baseQuery:"query1"}}
      override={true}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("HeaderBar - should create snapshot (lineup)", () => {
    const component = renderer.create(<HeaderBar
      thisPage={ParamPrefixes.lineup}
      common={{team:"Test2", gender:"Men", year:"2019/20", baseQuery:"query2"}}
      override={true}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  test("HeaderBar - should create snapshot (report)", () => {
    const component = renderer.create(<HeaderBar
      thisPage={ParamPrefixes.report}
      common={{team:"Test3", gender:"Men", year:"2019/20", baseQuery:"query3"}}
      override={true}
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
