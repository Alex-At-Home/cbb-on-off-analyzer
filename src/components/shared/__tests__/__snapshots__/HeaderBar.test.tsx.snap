// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar HeaderBar - should create snapshot (game) 1`] = `
<div
  className="container"
>
  <div
    className="border-top row"
  >
    <div
      className="text-center small col"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <a
          href=""
          onClick={[Function]}
        >
          Leaderboards
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-down fa-w-10 "
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="text-center small col"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <a
          href=""
          onClick={[Function]}
        >
          Lineups
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-down fa-w-10 "
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="text-center small col"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <a
          href=""
          onClick={[Function]}
        >
          Reports
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-down fa-w-10 "
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="text-center small col"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <a
          href=""
          onClick={[Function]}
        >
          Charts
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-down fa-w-10 "
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="small col-1"
    >
      <div>
        <a
          href="https://hoop-explorer.blogspot.com/p/blog-page.html"
          target="_blank"
        >
          Blog...
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderBar HeaderBar - should create snapshot (lineup) 1`] = `
<div
  className="container"
>
  <div
    className="border-top row"
  >
    <div
      className="text-center small col"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <a
          href=""
          onClick={[Function]}
        >
          Leaderboards
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-down fa-w-10 "
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="text-center small col"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <a
          href=""
          onClick={[Function]}
        >
          On/Off
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-down fa-w-10 "
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="text-center small col"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <a
          href=""
          onClick={[Function]}
        >
          Reports
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-down fa-w-10 "
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="text-center small col"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <a
          href=""
          onClick={[Function]}
        >
          Charts
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-down fa-w-10 "
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="small col-1"
    >
      <div>
        <a
          href="https://hoop-explorer.blogspot.com/p/blog-page.html"
          target="_blank"
        >
          Blog...
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeaderBar HeaderBar - should create snapshot (report) 1`] = `
<div
  className="container"
>
  <div
    className="border-top row"
  >
    <div
      className="text-center small col"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <a
          href=""
          onClick={[Function]}
        >
          Leaderboards
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-down fa-w-10 "
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="text-center small col"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <a
          href=""
          onClick={[Function]}
        >
          On/Off
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-down fa-w-10 "
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="text-center small col"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <a
          href=""
          onClick={[Function]}
        >
          Lineups
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-down fa-w-10 "
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="text-center small col"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <a
          href=""
          onClick={[Function]}
        >
          Reports
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-down fa-w-10 "
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="text-center small col"
    >
      <div
        className="dropdown"
        onKeyDown={[Function]}
      >
        <a
          href=""
          onClick={[Function]}
        >
          Charts
            
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-angle-down fa-w-10 "
            data-icon="angle-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </a>
      </div>
    </div>
    <div
      className="small col-1"
    >
      <div>
        <a
          href="https://hoop-explorer.blogspot.com/p/blog-page.html"
          target="_blank"
        >
          Blog...
        </a>
      </div>
    </div>
  </div>
</div>
`;
